benchmarks/trace/generate.go

```diff
@@ -0,0 +1,161 @@
+package trace
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"math"
+	"net"
+	"os"
+	"path/filepath"
+	"testing"
+	"text/template"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	"go.opencensus.io/benchmarks/defs"
+	"go.opencensus.io/plugin/ocgrpc"
+
+	"github.com/odeke-em/cli-spinner"
+)
+
+var addr = ":8796"
+
+var tmpl = template.Must(template.New("chartIt.html").
+	Funcs(template.FuncMap{
+		"json": func(v interface{}) string {
+			blob, _ := json.Marshal(v)
+			return string(blob)
+		},
+	}).Parse(chartHTML))
+
+func Generate(dir string) {
+	ln, err := net.Listen("tcp", addr)
+	if err != nil {
+		log.Fatalf("Listening on addr %q err: %v", addr, err)
+	}
+	srv := grpc.NewServer(grpc.StatsHandler(ocgrpc.NewServerStatsHandler()))
+	defs.RegisterPingServer(srv, new(server))
+	go srv.Serve(ln)
+	defer srv.Stop()
+
+	metrics := runTestsAndCollectMetrics()
+	fileCreator := map[string]bool{
+		"allocs":    true,
+		"througput": false,
+	}
+
+	for key, graphingAllocs := range fileCreator {
+		outFile := filepath.Join(dir, fmt.Sprintf("%s.html", key))
+		f, err := os.Create(outFile)
+		if err != nil {
+			log.Fatalf("Creating graph display file: %q err: %v", key, err)
+		}
+		metrics.GraphingAllocs = graphingAllocs
+		if err := tmpl.Execute(f, metrics); err != nil {
+			log.Fatalf("Creating template: %v", err)
+		}
+		f.Close()
+		log.Printf("Generated file: %q", outFile)
+	}
+}
+
+type server int
+
+func (s *server) Ping(ctx context.Context, p *defs.Payload) (*defs.Payload, error) {
+	return &defs.Payload{"Pong"}, nil
+}
+
+type graphData struct {
+	QPS        []string
+	MemAllocs  []float64
+	Throughput []float64
+}
+
+type graphTemplate struct {
+	GraphingAllocs bool
+	Traced         *graphData
+	Untraced       *graphData
+}
+
+func runTestsAndCollectMetrics() *graphTemplate {
+	log.Printf("Running tests to collect the metrics")
+	spin := spinner.New(10)
+	spin.Start()
+	defer spin.Stop()
+
+	graphingMap := make(map[string]*graphData)
+
+	funcsMap := map[string]func(*testing.B, int){
+		"Traced":   benchmarkTraced,
+		"Untraced": benchmarkUntraced,
+	}
+
+	qpsL := []int{1, 10, 100, 1000}
```

Comment:
Can we either have this benchmark or the trace/benchmarks.go? We are implementing the same benchmarks twice. I'd say we don't need visualization right now, let's leave this out for a while.